tools/c7n_gcp/c7n_gcp/query.py

```diff
@@ -161,13 +166,68 @@ def resources(self, query=None):
                     local_session(self.session_factory).get_default_project())
                 return []
             raise
-        self._cache.save(key, resources)
-        return self.filter_resources(resources)
 
     def augment(self, resources):
         return resources
 
 
+class ChildResourceManager(QueryResourceManager):
+
+    @staticmethod
+    def _get_parent_resource_info(resource_info, child_resource_type, parent_resource_type):
+        parent_resource_info = {}
+        parent_resource_info.update(resource_info)
+
+        arg_name = child_resource_type.parent_spec['arg_name']
+        parent_resource_info[parent_resource_type.id] = resource_info[arg_name]
+
+        return parent_resource_info
+
+    def get_resource(self, resource_info):
+        child_instance = super(ChildResourceManager, self).get_resource(resource_info)
+
+        parent_resource = self.resource_type.parent_spec['resource']
+        parent_resource_manager = self.get_resource_manager(parent_resource)
+
+        parent_resource_info = self._get_parent_resource_info(
+            resource_info, self.resource_type, parent_resource_manager.resource_type
+        )
+
+        parent_instance = parent_resource_manager.get_resource(parent_resource_info)
+
+        annotation_key = self.resource_type.get_parent_annotation_key()
+        child_instance[annotation_key] = parent_instance
+
+        return child_instance
+
+    def _fetch_resources(self, query):
+        if not query:
+            query = {}
+
+        resources = []
+
+        parent_spec = self.resource_type.parent_spec
+        parent_resource = parent_spec['resource']
```

Comment:
stylistic nit: there's a lot of temp variable aliasing in the pr, if its only going to be used once then just drop the temp var.

Reply:
Dropped where it didn't cause the long line lint failure